tools/populate-owners/main_test.go

```diff
@@ -0,0 +1,484 @@
+							"ab": {
+								"alice",
+								"bob",
+							},
+						},
+					},
+				},
+			},
+			namespaced: []*orgRepo{
+				{
+					Organization: "a",
+					Repository:   "b",
+					Aliases: &aliases{
+						Aliases: map[string][]string{
+							"ab": {
+								"alice",
+								"bob",
+							},
+						},
+					},
+				},
+				{
+					Organization: "c",
+					Repository:   "d",
+					Aliases: &aliases{
+						Aliases: map[string][]string{
+							"ab": {
+								"alice",
+								"bob",
+							},
+						},
+					},
+				},
+			},
+			collected: &aliases{
+				Aliases: map[string][]string{
+					"ab": {
+						"alice",
+						"bob",
+					},
+				},
+			},
+		},
+		{
+			name: "different members",
+			input: []*orgRepo{
+				{
+					Organization: "a",
+					Repository:   "b",
+					Aliases: &aliases{
+						Aliases: map[string][]string{
+							"ab": {
+								"alice",
+								"bob",
+							},
+						},
+					},
+				},
+				{
+					Organization: "c",
+					Repository:   "d",
+					Aliases: &aliases{
+						Aliases: map[string][]string{
+							"ab": {
+								"bob",
+								"charlie",
+							},
+						},
+					},
+				},
+			},
+			namespaced: []*orgRepo{
+				{
+					Organization: "a",
+					Repository:   "b",
+					Aliases: &aliases{
+						Aliases: map[string][]string{
+							"a-b-ab": {
+								"alice",
+								"bob",
+							},
+						},
+					},
+				},
+				{
+					Organization: "c",
+					Repository:   "d",
+					Aliases: &aliases{
+						Aliases: map[string][]string{
+							"c-d-ab": {
+								"bob",
+								"charlie",
+							},
+						},
+					},
+				},
+			},
+			collected: &aliases{
+				Aliases: map[string][]string{
+					"a-b-ab": {
+						"alice",
+						"bob",
+					},
+					"c-d-ab": {
+						"bob",
+						"charlie",
+					},
+				},
+			},
+		},
+		{
+			name: "collision",
+			input: []*orgRepo{
+				{
+					Organization: "a",
+					Repository:   "b",
+					Aliases: &aliases{
+						Aliases: map[string][]string{
+							"ab": {
+								"alice",
+								"bob",
+							},
+						},
+					},
+				},
+				{
+					Organization: "c",
+					Repository:   "d",
+					Aliases: &aliases{
+						Aliases: map[string][]string{
+							"ab": {
+								"bob",
+								"charlie",
+							},
+						},
+					},
+				},
+				{
+					Organization: "e",
+					Repository:   "f",
+					Aliases: &aliases{
+						Aliases: map[string][]string{
+							"a-b-ab": {
+								"bob",
+								"charlie",
+							},
+						},
+					},
+				},
+			},
+			error: regexp.MustCompile("^namespaced alias collision: \"a-b-ab\"$"),
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			collected, err := namespaceAliases(test.input)
+			if test.error == nil {
+				if err != nil {
+					t.Fatal(err)
+				}
+			} else if !test.error.MatchString(err.Error()) {
+				t.Fatalf("unexpected error: %v does not match %v", err, test.error)
+			}
+
+			assertEqual(t, collected, test.collected)
+			if test.namespaced != nil {
```

Comment:
wait. is `namespaceAliases` mutating the input?

Reply:
either let's add godoc to explain the data flow or let's not mutate the input and just have clear input --> output flow

Reply:
> either let's add godoc to explain the data flow...

Added with a73bc68 -> f28cfdf.